test/reshape.jl

```diff
@@ -348,4 +348,32 @@ end
     @test wide3 == w2[!, Not(2)]
 end
 
+@testset "flatten" begin
+    df_vec = DataFrame(a = [1, 2], b = [[1, 2], [3, 4]])
+    df_tup = DataFrame(a = [1, 2], b = [(1, 2), (3, 4)])
+    ref = DataFrame(a = [1, 1, 2, 2], b = [1, 2, 3, 4])
+    @test flatten(df_vec, :b) == flatten(df_tup, :b) == ref
+    df_mixed_types = DataFrame(a = [1, 2], b = [[1, 2], ["x", "y"]])
+    ref_mixed_types = DataFrame(a = [1, 1, 2, 2], b = [1, 2, "x", "y"])
+    @test flatten(df_mixed_types, :b) == ref_mixed_types
+    df_three = DataFrame(a = [1, 2, 3], b = [[1, 2], [10, 20], [100, 200, 300]])
+    ref_three = DataFrame(a = [1, 1, 2, 2, 3, 3, 3], b = [1, 2, 10, 20, 100, 200, 300])
+    @test flatten(df_three, :b) == ref_three
+    df_gen = DataFrame(a = [1, 2], b = [(i for i in 1:5), (i for i in 6:10)])
+    ref_gen = DataFrame(a = [fill(1, 5); fill(2, 5)], b = collect(1:10))
+    @test flatten(df_gen, :b) == ref_gen
+    df_miss = DataFrame(a = [1, 2], b = [Union{Missing, Int}[1, 2], Union{Missing, Int}[3, 4]])
+    ref = DataFrame(a = [1, 1, 2, 2], b = [1, 2, 3, 4])
+    @test flatten(df_miss, :b) == ref
+    v1 = [[1, 2], [3, 4]]
+    v2 = [[5, 6], [7, 8]]
+    v = [v1, v2]
+    df_vec_vec = DataFrame(a = [1, 2], b = v)
+    ref_vec_vec = DataFrame(a = [1, 1, 2, 2], b = [v1 ; v2])
+    @test flatten(df_vec_vec, :b) == ref_vec_vec
+    df_cat = DataFrame(a = [1, 2], b = [CategoricalArray([1, 2]), CategoricalArray([1, 2])])
+    ref_cat = DataFrame(a = [1, 1, 2, 2], b = CategoricalArray([1, 2, 1, 2]))
```

Comment:
Did you mean this to be `ref`? Anyway, there's no need to make `b` categorical as it doesn't make any difference for `==`. But you have to test that the resulting column is a categorical array.